File: 3309-CountPrefixAndSuffixPairsI/3309-CountPrefixAndSuffixPairsI.java
// Last updated: 31/07/2025, 21:05:48
class Solution {
    public int countPrefixSuffixPairs(String[] words) {
        int l=words.length;
        int c=0;
        for(int i=0;i<l-1;i++)
        {
            for(int j=i+1;j<l;j++)
            {
                int l1=words[i].length();
                int l2=words[j].length();
                if(l1<=l2)
                {

                
                    if(words[i].equals(words[j].substring(0,l1)) && words[i].equals(words[j].substring(l2-l1)))
                {
                    c++;
                }
            }
        }
        }
    return c;
        
    }
}

File: 3309-CountPrefixAndSuffixPairsI/3309-CountPrefixAndSuffixPairsI.py
# Last updated: 31/07/2025, 21:05:41
class Solution:
    def countPrefixSuffixPairs(self, words: List[str]) -> int:
        c=0
        for i in range(len(words)-1):
            for j in range(i+1,len(words)):
                l1=len(words[i])
                l2=len(words[j])
                if l1<=l2:
                    if words[i]==words[j][0:l1] and words[i]==words[j][l2-l1:]:
                        c+=1
        return c                
        